lib/internal/errors.js

```diff
@@ -812,11 +812,21 @@ E('ERR_INVALID_SYNC_FORK_INPUT',
   TypeError);
 E('ERR_INVALID_THIS', 'Value of "this" must be of type %s', TypeError);
 E('ERR_INVALID_TUPLE', '%s must be an iterable %s tuple', TypeError);
-E('ERR_INVALID_TYPE_EXTENSION', '%s extension is not supported for --type=%s',
-  TypeError);
+E('ERR_INVALID_TYPE_EXTENSION', (extension, flagValue) => {
+  if (flagValue === 'module')
+    flagValue = 'module or -m';
+  return `${extension} is not supported for --type=${flagValue}`;
+}, TypeError);
 E('ERR_INVALID_TYPE_FLAG',
-  'Type flag must be one of "esm", "commonjs". Received --type=%s',
+  'Type flag must be one of "module", "commonjs". Received --type=%s',
   TypeError);
+E('ERR_INVALID_TYPE_IN_PACKAGE_SCOPE', (flagValue, scopeType) => {
+  if (flagValue === 'module')
+    flagValue = 'module or -m';
+  return `Cannot use --type=${flagValue} because nearest parent package.json` +
+  ` includes "type": "${scopeType}"${(scopeType === 'commonjs') ?
+    ' (or lacks the "type" field)' : ''}`;
+}, TypeError);
```

Comment:
I was thinking it might be nice to unify these errors under a single `UNEXPECTED_MODULE_FORMAT` or similar.

Reply:
Note we also need to include the spec changes for this in the docs. Will see if I can push that too.